### ISSUE TITLE
XML Mapping: dimensions must be a string instead an array

### ISSUE THREAD

**Issue by chmelik** [dustin10/VichUploaderBundle#972]
### Bug Report

Error when use "dimensions" for mapping.

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 1.8.6

#### Summary

```
Type error: Argument 2 passed to Vich\UploaderBundle\Mapping\PropertyMapping::fixPropertyPath() must be of the type string, array given, called in ./vich/uploader-bundle/Mapping/PropertyMapping.php on line 191
```

Because #914 was fixed by https://github.com/dustin10/VichUploaderBundle/commit/2f4cecc9a88dd9572b665132f28bc9515cbe3af8

```
'dimensions' => (array) $field->attributes()->dimensions,
```

.. but need "string"

**Comment by garak** [comment]
Indeed, `$this->propertyPaths[$property]` should resolve to a string.

**Comment by garak** [comment]
Closing for missing feedback

**Comment by chmelik** [comment]
> Indeed, `$this->propertyPaths[$property]` should resolve to a string.

yes, I thought you confirm it)
we except "string" and must return "string": https://github.com/dustin10/VichUploaderBundle/blob/1.8.6/Mapping/PropertyMapping.php#L373
..but pass "array"

**Comment by garak** [comment]
I'm not sure to get you

**Comment by chmelik** [comment]
when we see on https://github.com/dustin10/VichUploaderBundle/blob/1.8.6/Metadata/Driver/XmlDriver.php#L26-L34
all fields are converted to "string", but line 33
```
'dimensions' => null === $field->attributes()->dimensions ? null : (array) $field->attributes()->dimensions,
```
to "array"

and later we pass this value to 
in https://github.com/dustin10/VichUploaderBundle/blob/1.8.6/Mapping/PropertyMapping.php#L373

so, to fix it, we need just resolve to "string"
```
'dimensions' => null === $field->attributes()->dimensions ? null : (string) $field->attributes()->dimensions,
```
in /Metadata/Driver/XmlDriver.php#L26-L34

**Comment by garak** [comment]
But if we resolve dimensions to string instead of array, the bug we closed with previous change will raise again.

**Comment by chmelik** [comment]
No, if we look to https://github.com/dustin10/VichUploaderBundle/commit/2f4cecc9a88dd9572b665132f28bc9515cbe3af8 expression "$field->attributes()->dimensions" returned SimpleXMLElement object and for fix that we need resolve to "string" (like other fields in XmlDriver.php#L26-L34), but not to "array".
.. all fields must be a string or null when not configured (like https://github.com/dustin10/VichUploaderBundle/commit/63ef0ea6d9d001e5110f4a647b4dd21fce18fd70 and in YamlDriver.php) (https://github.com/dustin10/VichUploaderBundle/blob/1.8.6/Metadata/Driver/XmlDriver.php#L26-L34).

as result $this->propertyPaths[$property] always will be a string as we expect (and serialization in https://github.com/dustin10/VichUploaderBundle/issues/914 will be work well).

and metadata from YamlDriver and XmlDriver will be identical.

**Comment by garak** [comment]
Can you arrange a PR?

**Comment by garak** [comment]
Fixed in #985
